### PR TITLE
Added test backpressure published fn

### PR DESCRIPTION
**Pull request by gnought**

### CODE REVIEW

File: test/events.js [hunk 1]
```diff
@@ -122,3 +122,43 @@ test('Emit event when broker closed', function (t) {
...
 })
+
+test('Test backpressure aedes published function', { timeout: 500 }, function (t) {
+  t.plan(1)
+
+  const broker = aedes({
+    published: function (packet, client, done) {
+      if (client) {
+        t.pass('published')
+        setTimeout(done, 1000)
```
Comment (mcollina) [inline test/events.js]: I don't understand how this timeout relates to the timeout in the actual test.

File: test/events.js [hunk 2]
```diff
@@ -122,3 +122,43 @@ test('Emit event when broker closed', function (t) {
...
+  })
+  t.tearDown(() => {
+    client.end()
+    broker.close()
+    server.close()
+  })
+
+  const mqtt = require('mqtt')
+  const server = require('net').createServer(broker.handle)
+  const port = 1883
```
Comment (mcollina) [inline test/events.js]: please set this to 0 and get the `port` from `server.address().port`

File: test/events.js (hunk 1 above)
Comment (gnought) [reply test/events.js]: this `setTimeout` is to stimulate a backpressure on published function. The whole test timeout is set less than 1000 to let the test plan end with 1 passed result.

Comment (mcollina) [reply test/events.js]: How long will the test last?

Comment (gnought) [reply test/events.js]: 500

File: test/events.js [hunk 3]
```diff
@@ -123,30 +123,33 @@ test('Emit event when broker closed', function (t) {
...
   const broker = aedes({
     published: function (packet, client, done) {
       if (client) {
         t.pass('published')
-        setTimeout(done, 1000)
+        setTimeout(() => {
+          publisher.end()
+          done()
+        }
+        , 1000)
```
Comment (mcollina) [inline test/events.js]: can you remove this?

File: test/events.js [hunk 4]
```diff
@@ -122,3 +122,45 @@ test('Emit event when broker closed', function (t) {
...
+
+  const mqtt = require('mqtt')
+  const server = require('net').createServer(broker.handle)
+  var publisher
+
+  server.listen(0, function () {
+    const port = server.address().port
+    publisher = mqtt.connect({ port: port, host: 'localhost', clean: true, keepalive: 30 })
+
+    var count = 10
```
Comment (mcollina) [inline test/events.js]: We should verify that count reach a certain value when the test ends, maybe 0?

**Pull request merged**